app/waf/test/acceptance/wafpolicy.controller.acceptance.ts

```diff
@@ -70,6 +95,142 @@ describe('WafpolicyController', () => {
+        .set('tenant-id', ExpectedData.tenantId)
+        .send()
+        .expect(204);
+    },
+  );
+
+  it(
+    'post ' +
+      prefix +
+      '/wafpolicies/${id}/adcs/${adcId}: uploading wafpolicy with a private wafpolicy',
+    async () => {
+      const wafpolicy = await givenWafpolicyData(wafapp, {
+        tenantId: ExpectedData.tenantId,
+        public: false,
+      });
+
+      const adc = await givenAdcData(wafapp, {
+        tenantId: ExpectedData.tenantId,
+        trustedDeviceId: uuid(),
+      });
+
+      uploadWafpolicyStub.returns({
+        id: uuid(),
+        name: `${wafpolicy.id}`,
+        enforcementMode: 'blocking',
+        lastChanged: 'random',
+        lastChange: 'random',
+        state: 'UPLOADING',
+        path: `/Common/${wafpolicy.id}`,
+      });
+
+      await client
+        .post(prefix + `/wafpolicies/${wafpolicy.id}/adcs/${adc.id}`)
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .send()
+        .expect(204);
+    },
+  );
+
+  it(
+    'post ' +
+      prefix +
+      '/wafpolicies/${id}/adcs/${adcId}: uploading wafpolicy with an untrunsted device',
+    async () => {
+      const wafpolicy = await givenWafpolicyData(wafapp, {
+        tenantId: ExpectedData.tenantId,
+        public: false,
+      });
+
+      const adc = await givenAdcData(wafapp, {
+        tenantId: ExpectedData.tenantId,
+      });
+
+      uploadWafpolicyStub.returns({
+        id: uuid(),
+        name: `${wafpolicy.id}`,
+        enforcementMode: 'blocking',
+        lastChanged: 'random',
+        lastChange: 'random',
+        state: 'UPLOADING',
+        path: `/Common/${wafpolicy.id}`,
+      });
+
+      await client
+        .post(prefix + `/wafpolicies/${wafpolicy.id}/adcs/${adc.id}`)
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .send()
+        .expect(422);
+    },
+  );
+
+  it(
+    'get ' + prefix + '/wafpolicies/${id}/adcs/${adcId}: get wafpolicy status',
+    async () => {
+      const wafpolicy = await givenWafpolicyData(wafapp, {
+        tenantId: 'a random id',
+        public: true,
+      });
+
+      const adc = await givenAdcData(wafapp, {
+        tenantId: ExpectedData.tenantId,
+        trustedDeviceId: uuid(),
+      });
+
+      checkWafpolicyStub.returns([
+        {
+          id: uuid(),
+          name: `${wafpolicy.id}`,
+          enforcementMode: 'blocking',
+          lastChanged: 'random',
+          lastChange: 'random',
+          state: 'UPLOADING',
+          path: `/Common/${wafpolicy.id}`,
+        },
+      ]);
+
+      const resp = await client
+        .get(prefix + `/wafpolicies/${wafpolicy.id}/adcs/${adc.id}`)
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .expect(200);
+
+      expect(resp.body.wafpolicyondevice.state).equal('UPLOADING');
```

Comment:
Will you check `UPLOADED`? What's the final state.

Reply:
This is a test, the response is stubbed by myself, it does not matter what the state it is.

Reply:
Got it.